packages/api/core/src/util/forge-config.ts

```diff
@@ -123,9 +124,11 @@ export default async (dir: string): Promise<ResolvedForgeConfig> => {
 
   if (await forgeConfigIsValidFilePath(dir, forgeConfig)) {
     try {
+      // The loaded "config" could potentially be a static forge config, ESM module or async function
       // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig>;
-      forgeConfig = 'default' in loaded ? loaded.default : loaded;
+      const loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig> | AsyncFn;
```

Comment:
```suggestion
      const loaded = require(path.resolve(dir, forgeConfig as string)) as MaybeESM<ForgeConfig | AsyncForgeConfig>;
```

I think that's more technically accurate, surprised TS is happy with it the other way round

Reply:
Maybe the union typing threw it off, thinking we'd accept either `MaybeESM<T>` or the function type? This new way is more accurate though, you're right - fixed 👍